tests/integration/java-integration/src/test/java/io/fluentbit/test/IntegrationTest.java

```diff
@@ -0,0 +1,153 @@
+/*  Fluent Bit
+ *  ==========
+ *  Copyright (C) 2019-2021 The Fluent Bit Authors
+ *  Copyright (C) 2015-2018 Treasure Data Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package io.fluentbit.test;
+
+import org.junit.Ignore;
+import org.junit.Test;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import static io.fluentbit.test.Util.sleep;
+import static io.fluentbit.test.Util.waitFor;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * This class runs the actual tests.
+ */
+public class IntegrationTest {
+
+
+    /**
+     * A very simple test. One message is send, all services are up and behaving.
+     */
+    @Test
+    public void happyPathOneMessage() throws Exception {
+        ExpectedLogData expectedLogData = new ExpectedLogData(1);
+        try (LoggingDestinationServer loggingDestinationServer = new LoggingDestinationServer(expectedLogData)) {
+            try (FluentBit fluentBit = new FluentBit()) {
+                try (MyLogger logger = new MyLogger(expectedLogData)) {
+                    logger.log("message");
+                }
+                waitForMessagesProcessed(TimeUnit.SECONDS, 15, expectedLogData);
```

Comment:
I'm not a fan of timeouts as they're arbitrary and usually fail in CI or other people's machines with different timings. It should be a very upper bound "there is no way it can be longer than this" with tests completing beforehand ideally if successful.

Can you clarify what the timeout is for here? Could we have one per message and trigger a failure sooner - goal is to fail fast I think if we can.

Reply:
OK, as long as that is plenty - the Github runners can be a little slow as they're multitenant overprovisioned sometimes.